src/telemetry/net_if_properties.rs

```diff
@@ -0,0 +1,125 @@
+/*
+ * This file is part of Edgehog.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use crate::error::DeviceManagerError;
+use astarte_sdk::types::AstarteType;
+use log::warn;
+use std::collections::HashMap;
+
+#[derive(Debug)]
+enum TechnologyType {
+    Ethernet,
+    Cellular,
+    WiFi,
+}
+
+impl ToString for TechnologyType {
+    fn to_string(&self) -> String {
+        match self {
+            TechnologyType::Ethernet => "Ethernet",
+            TechnologyType::Cellular => "Cellular",
+            TechnologyType::WiFi => "WiFi",
+        }
+        .to_owned()
+    }
+}
+
+#[derive(Debug)]
+struct NetworkInterfaceProperties {
+    interface: String,
+    mac_address: String,
+    technology_type: TechnologyType,
+}
+
+fn get_ethernet_wifi() -> Result<Vec<NetworkInterfaceProperties>, DeviceManagerError> {
+    let mut results = Vec::new();
+
+    let mut enumerator = udev::Enumerator::new()?;
+
+    enumerator.match_subsystem("net")?;
+
+    for device in enumerator.scan_devices()? {
+        let (address, type_) = match (
+            device.attribute_value("address"),
+            device.attribute_value("type"),
+        ) {
+            (Some(addr), Some(type_)) => (addr, type_),
+            _ => continue,
+        };
+
+        let technology_type = match type_.to_string_lossy().trim() {
+            "1" => {
```

Comment:
Maybe here you can use a const specifying the real name.
``` rust
const ARPHRD_ETHER:i32 = 1;
const ARPHRD_PPP:i32 = 512;
```
Ref: https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/include/uapi/linux/if_arp.h?h=linux-4.14.y#n31